src/pages/messages/messages.ts

```diff
@@ -172,11 +192,28 @@ export class MessagesPage implements OnInit, OnDestroy {
     });
   }
 
-  sendMessage(): void {
+  showPicture({ target }: Event) {
+    const modal = this.modalCtrl.create(ShowPictureComponent, {
+      pictureSrc: (<HTMLImageElement>target).src
+    });
+
+    modal.present();
+  }
+
+  sendPictureMessage(file: File): void {
+    this.pictureService.upload(file).then((picture) => {
+      MeteorObservable.call('addMessage', 'picture',
+        this.selectedChat._id,
+        picture.url
+      ).zone().subscribe();
+    });
+  }
+
+  sendTextMessage(): void {
     // If message was yet to be typed, abort
     if (!this.message) return;
 
-    MeteorObservable.call('addMessage',
+    MeteorObservable.call('addMessage', 'text',
```

Comment:
Please move `text` type into an enum of MessageType (media, text and I will later add location(